### ISSUE TITLE
Unable to locate mscorlib

### ISSUE THREAD

**Issue by yhax** [aspnet/Mvc#2991]
I'm trying to add a c# class library (4.5) to my MVC 6 class library. Pre beta 6 & RTM, we had been using nuget to get the library into the project. No problems.

Since beta 6 / RTM we have been running into an issue where npm says it is unable to locate mscorlib.
I've removed the whole nuget package thing, and added the 4.5 class library directly to my solution. The class library builds fine, it's pretty basic, and has the following references:

![image](https://cloud.githubusercontent.com/assets/3464157/9332315/0b024936-4592-11e5-8b0d-714101784868.png)

If I leave my project.json file empty, a few libraries are automatically added, mscorlib is one of them (and resolves correctly) -

![image](https://cloud.githubusercontent.com/assets/3464157/9332403/6ee1d91c-4592-11e5-81ab-a1fb56bf8b6a.png)

The issue comes when I try to add the 4.5 class library -

![image](https://cloud.githubusercontent.com/assets/3464157/9332470/b016c730-4592-11e5-85a7-f0c3cfb0e9da.png)

Errors in ....project.json
    Unable to locate mscorlib >= 4.0.0.0
Microsoft .NET Development Utility CLR-x86-1.0.0-beta6-12256

It automatically adds the dependency in the project.lock.json file. If I remove it, and lock the project.lock.json file, it works correctly. However since this is being used by a bunch of projects, locking is an issue.

Any ideas? thanks!

**Comment by danroth27** [comment]
@yhax Can you see if the project builds at the command-line by running `dnu build` from the VS developer command prompt in the same folder as the project.json file?

**Comment by yhax** [comment]
@danroth27 sure. So I created a brand new class library project. I tried building without adding the 4.5 library. This is the output:

![a](https://cloud.githubusercontent.com/assets/3464157/9365475/bfd196c6-467f-11e5-931b-91b38da83863.jpg)

I then added the 4.5 project to the solution -

![b](https://cloud.githubusercontent.com/assets/3464157/9365483/cca37b12-467f-11e5-9b85-3ebc12866045.jpg)

And then I tried the dnu build using the VS 2015 dev cmd prompt -

![c](https://cloud.githubusercontent.com/assets/3464157/9365495/df31a31c-467f-11e5-9468-097a4e736df6.jpg)

(The message cut off says - Dependency specified was Contracts.V5 >= 1.0.0-* but ended up with Contracts.V5 45.5.0-beta.

**Comment by davidfowl** [comment]
What framework does `Contracts.V5 45.5.0-beta` target?

**Comment by yhax** [comment]
@davidfowl it targets .NET Framework 4.5. I also tried 4.6, with the same results. 

One thing I failed to mention. The contracts library was created with VS 2013. Not sure if that matters. I tried creating an empty library and importing it into the 2015 project above. I don't get the mscorlib problem, but it tells me it can't locate the library:

![image](https://cloud.githubusercontent.com/assets/3464157/9440663/d8ddb0a6-4a3e-11e5-84e1-6c66a6a4e626.png)

![image](https://cloud.githubusercontent.com/assets/3464157/9440687/02642a36-4a3f-11e5-9297-32d6cc369523.png)
